globus_sdk/config/env_vars.py

```diff
@@ -0,0 +1,68 @@
+"""
+Definition and loading of standard environment variables, plus a wrappers for loading
+and parsing values.
+"""
+import logging
+import os
+import typing
+
+log = logging.getLogger(__name__)
+
+
+ENVNAME_VAR = "GLOBUS_SDK_ENVIRONMENT"
+HTTP_TIMEOUT_VAR = "GLOBUS_SDK_HTTP_TIMEOUT"
+SSL_VERIFY_VAR = "GLOBUS_SDK_VERIFY_SSL"
+
+
+def _load_var(
+    varname: str, default: str, explicit_value: typing.Optional[str] = None
+) -> str:
+    # use the explicit value if given and non-None, otherwise, do an env lookup
+    value = (
+        explicit_value if explicit_value is not None else os.getenv(varname, default)
+    )
+    # only info log on non-default *values*
+    # meaning that if we define the default as 'foo' and someone explicitly sets 'foo',
+    # no info log gets emitted
+    if value != default:
+        log.info(f"on lookup, non-default setting: {varname}={value}")
+    else:
+        log.debug(f"on lookup, default setting: {varname}={value}")
+    return value
+
+
+def _bool_cast(value: str) -> bool:
+    value = value.lower()
+    if value in ("1", "yes", "true", "on"):
+        return True
+    elif value in ("0", "no", "false", "off"):
+        return False
+    log.error(f'Value "{value}" can\'t cast to bool')
+    raise ValueError(f"Invalid config bool: {value}")
```

Comment:
I think this could maybe use [strtobool](https://docs.python.org/3/distutils/apiref.html#distutils.util.strtobool) which has an almost identical behavior.

```suggestion
from distutils.utils import strtobool

def _bool_cast(value: str) -> bool:
    return strtobool(value) == 1
```

Reply:
I'll need to adjust this slightly, since I redid things a bit when handling `http_timeout`, but I'll incorporate this.
I feel a slight aversion to depending on `distutils`, but this really doesn't matter much and it's easy to rewrite it in any way we see fit.

Reply:
Committed as 92e3d5d
(Looks like I didn't do the co-authored-by line quite right, but I'm not sure exactly why that's not working. If you care, we can amend.)